src/Libraries/Microsoft.Extensions.TimeProvider.Testing/FakeTimeProvider.cs

```diff
@@ -14,13 +15,16 @@ namespace Microsoft.Extensions.Time.Testing;
 /// </summary>
 public class FakeTimeProvider : TimeProvider
 {
-    internal static readonly DateTimeOffset Epoch = new(2000, 1, 1, 0, 0, 0, 0, TimeSpan.Zero);
-
     internal readonly List<FakeTimeProviderTimer.Waiter> Waiters = new();
 
-    private DateTimeOffset _now = Epoch;
+    private DateTimeOffset _now;
     private TimeZoneInfo _localTimeZone;
 
+    /// <summary>
+    /// Gets the time at which the clock was initialized.
+    /// </summary>
+    public DateTimeOffset Epoch { get; } = new DateTimeOffset(2000, 1, 1, 0, 0, 0, 0, TimeSpan.Zero);
```

Comment:
I don't think `Epoch` should be changed or made `public` or replace `startTime`. I don't think it's related to the PR, and I don't understand the reason why it would, can you explain the reasons?

Reply:
Saw the reasons in the comments sorry. I'd suggest to handle that separately or this could slow down the review. And not use the word `epoch` instead of start time, it means different things for me. "Epoch" should be a constant by definition.

Reply:
I can separate it out if you prefer, but I also used it in some of the tests for the central part of this feature, which was why I included it in this PR.

As for the name, I'm not a native English speaker, and my initial name for the property was also "StartTime". 

However, similar to us not using method name "AdvanceTime()" or "ForwardTime()", where the word "Time" is redundant/implicit due to the context it is used in, a "time provider" object, I started looking for alternatives and saw the word epoch in the original code. 

From the (oxford) dictionary, I see that epoch means _"the beginning of a period in the history of someone or something"_, and in that sense it makes sense to me that the `Epoch` property represents the beginning of time for the instance of the time provider.

All that said, I will leave it up to you and the team to decide on the final name. `StartTime`, `InitialUtcNow`, `StartUtcNow` are other alternatives I have been considering.

Reply:
> And not use the word `epoch` instead of start time, it means different things for me. "Epoch" should be a constant by definition.

I second that. I think it should be "start time" or something obvious like that.

> From the (oxford) dictionary, I see that epoch means _"the beginning of a period in the history of someone or something"_, and in that sense it makes sense to me that the `Epoch` property represents the beginning of time for the instance of the time provider.

I read it as the key being "_period in the history_", to me "epoch" is the same as an "era".

Reply:
I'm fine with that. I'll change it back to `StartTime`.

Two questions:

1. What about the internal field `epoch` that I removed. Do you want it back as it was or?

2. Do you want me to include the `StartTime` property in this PR or not?

Reply:
Renamed Epoch to StartTime.